### PR TITLE
Tests: Fix issues on Kafka AfterEach

### PR DESCRIPTION
**Pull request by eloycoto**
I saw the following error in the test logs:

```
time="2018-03-22T01:51:03Z" level=debug msg="running command: policy delete --all"
cmd: "policy delete --all" exitCode: 127
 bash: policy: command not found
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
Approving for the refactoring changes, but also want @manalibhutiyani 's approval before merging.

**Review by tgraf — APPROVED** [review]

File: test/runtime/kafka.go
Comment (manalibhutiyani) [inline test/runtime/kafka.go:205]: This does not need to be a separate background context. It will not cause an issue here hopefully,  but it was causing the earlier race we saw while producing/consuming to `allowedTopic`. Kafka always persists the messages, so we do not need to start the consumer beforehand, and  `--from-beginning` makes sure consumer reads from start of the persistent pipe.

Comment (manalibhutiyani) [reply test/runtime/kafka.go:205]: Also if it is *not* a separate background context, we will not require `err = res.WaitUntilMatch`

Comment (tgraf) [reply test/runtime/kafka.go:205]: What is a "separate background context"?

The change seems to do the right thing which is to match against the output of the consumer script to see the topic unauthorized output instead of assuming the the consumer script actually fails. It may retry for quite a while.

Comment (manalibhutiyani) [reply test/runtime/kafka.go:205]: I mean `ctx, cancel := context.WithCancel(context.Background())` is not required. The change itself is not incorrect. I was pointing out that earlier we had to explicitly put a sleep 10 after starting the consumer background task, because this "background consume task "  was not ready before we start producing to it (I might not be using the "correct" golang term here ...sorry, but I hope you get what I am trying to say) 

This is working because we are using a `res.WaitUntilMatch` 
and we are consuming on a disallowed topic.

Try this scenario:

1. ctx, cancel := context.WithCancel(context.Background()) for a consumer.
2. consume on ` allowedTopic` without waiting and expect a specific output *without* using `--from-beginning`
3. Produce a set of messages 
4. This will fail if we don't have some sort of wait/sleep between steps 1,2 because the "background task" takes a while to come up. This was also the reason we had the 10 sec sleep in all our earlier bash tests.
5. In this scenario, the `res.WaitUntilMatch` will not work either if  `--from-beginning` is not specified and we expect a specific output in consume.

Again, I am not a golang pro so excuse the incorrect usage of terms.

**Pull request merged**
